File: Domain/Models/Job.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace CitySurfing.Domain.Models
{
    public class Job : ModelBase
    {
        public Job()
        {
            IsAvailable = true;
            Created = DateTime.Now;
        }

        public string Title { get; set; }

        public string Description { get; set; }

        public double? Price { get; set; }

        public DateTime? StartDate { get; set; }

        public DateTime? EndDate { get; set; }

        public DateTime Created { get; set; }

        public string Location { get; set; }

        public bool IsAvailable { get; set; }

        [ForeignKey(nameof(Creator))]
        public string CreatorId { get; set; }

        [ForeignKey(nameof(Category))]
        public int CategoryId { get; set; }

        public virtual User Creator { get; set; }

        public virtual Category Category { get; set; }

        public virtual ICollection<Skill> RequiredSkills { get; set; } = new HashSet<Skill>();

        public virtual ICollection<Applyment> Applyments { get; set; } = new HashSet<Applyment>();

    }
}

File: app/src/app/cs/models/job.ts

import {Review} from './review';
import {User} from './user';
import {Skill} from './skill';
import {Category} from './category';
import {Applyment} from './applyment';

export class Job {
  Id: number;
  Title: string;
  Description: string;
  Price: number;
  StartDate: string;
  EndDate: string;
  Created: string;
  Location: string;
  IsAvailable: boolean;
  Review: Review;
  Creator: User;
  Type: Category;
  RequiredSkills: Skill[];
  Applyments: Applyment[];
}
